lambdas/checkpassport/src/test/java/uk/gov/di/ipv/cri/passport/checkpassport/services/DocumentDataVerificationServiceTest.java

```diff
@@ -0,0 +1,212 @@
+                .thenReturn(new ValidationResult<>(true, null));
+
+        when(mockThirdPartyAPIService.performCheck(passportFormData))
+                .thenReturn(thirdPartyAPIResult);
+
+        DocumentDataVerificationResult documentDataVerificationResult =
+                documentDataVerificationService.verifyData(passportFormData, sessionItem, null);
+
+        InOrder inOrder = inOrder(mockEventProbe);
+        inOrder.verify(mockEventProbe).counterMetric(FORM_DATA_VALIDATION_PASS);
+        inOrder.verify(mockEventProbe).counterMetric(DOCUMENT_DATA_VERIFICATION_REQUEST_SUCCEEDED);
+        verifyNoMoreInteractions(mockEventProbe);
+
+        verify(mockFormDataValidator).validate(passportFormData);
+        verify(mockThirdPartyAPIService).performCheck(passportFormData);
+
+        verify(mockAuditService)
+                .sendAuditEvent(eq(AuditEventType.REQUEST_SENT), any(AuditEventContext.class));
+        verify(mockAuditService)
+                .sendAuditEvent(
+                        eq(AuditEventType.RESPONSE_RECEIVED), any(AuditEventContext.class), eq(""));
+        verifyNoMoreInteractions(mockAuditService);
+
+        assertNotNull(documentDataVerificationResult);
+        assertTrue(documentDataVerificationResult.isVerified());
+        assertEquals(2, documentDataVerificationResult.getValidityScore());
+        assertEquals(4, documentDataVerificationResult.getStrengthScore());
+    }
+
+    @Test
+    void verifyIdentityShouldReturnValidationErrorWhenInvalidInputProvided() {
+        SessionItem sessionItem = new SessionItem();
+        sessionItem.setSessionId(UUID.randomUUID());
+
+        PassportFormData passportFormData = PassportFormTestDataGenerator.generate();
+
+        when(mockFormDataValidator.validate(passportFormData))
+                .thenReturn(new ValidationResult<>(false, List.of("validation error")));
+
+        OAuthHttpResponseExceptionWithErrorBody expectedReturnedException =
+                new OAuthHttpResponseExceptionWithErrorBody(
+                        HttpStatus.SC_INTERNAL_SERVER_ERROR,
+                        ErrorResponse.FORM_DATA_FAILED_VALIDATION);
+
+        OAuthHttpResponseExceptionWithErrorBody thrownException =
+                assertThrows(
+                        OAuthHttpResponseExceptionWithErrorBody.class,
+                        () -> {
+                            documentDataVerificationService.verifyData(
+                                    passportFormData, sessionItem, null);
+                        });
+
+        assertEquals(expectedReturnedException.getStatusCode(), thrownException.getStatusCode());
+        assertEquals(expectedReturnedException.getErrorReason(), thrownException.getErrorReason());
+        verify(mockEventProbe).counterMetric(FORM_DATA_VALIDATION_FAIL);
+        verify(mockEventProbe).counterMetric(DOCUMENT_DATA_VERIFICATION_REQUEST_FAILED);
+        verifyNoMoreInteractions(mockEventProbe);
```

Comment:
You could add your verifyNoMore Interactions to afterEach so you dont need to add them to each method individually